airflow/models/dag.py

```diff
@@ -1699,7 +1699,7 @@ def _get_task_instances(
         if include_subdags:
             # Crafting the right filter for dag_id and task_ids combo
             conditions = []
-            for dag in self.subdags + [self]:
+            for dag in [*self.subdags, self]:
```

Comment:
I wonder if `itertools.chain` would be better here

Reply:
I did some tests on it. It seems performance wise `[*[1, 2, 3], 4, 5, 6, 7]` is the best solution

```
$ python -m timeit '[1, 2, 3] + [4] + [5] + [6] + [7]'
2000000 loops, best of 5: 179 nsec per loop

$ python -m timeit '[*[1, 2, 3], 4, 5, 6, 7]'
5000000 loops, best of 5: 69.2 nsec per loop

$ python -m timeit 'import itertools; itertools.chain([1, 2, 3], [4, 5, 6, 7])'
2000000 loops, best of 5: 177 nsec per loop

$ python -m timeit 'import itertools; [1, 2, 3] + [4] + [5] + [6] + [7]'
1000000 loops, best of 5: 242 nsec per loop

$ python -m timeit 'import itertools;  [*[1, 2, 3], 4, 5, 6, 7]'
2000000 loops, best of 5: 131 nsec per loop

$ python -m timeit 'import itertools; list(itertools.chain([1, 2, 3], [4, 5, 6, 7]))'
1000000 loops, best of 5: 305 nsec per loop
```

Reply:
A big chunk of this would come from `import itertools` though, which would not be relevant in Airflow since the module is already imported in a lot of places. I would not be surprised if `*` is still best for small lists though, since the itertools version still needs to build an additional list (of a single item).